graphyte.py

```diff
@@ -169,12 +171,24 @@ def _thread_loop(self):
             if current_time - last_check_time >= self.interval:
                 last_check_time = current_time
                 if messages:
-                    self.send_socket(b''.join(messages))
+                    while messages:
+                        count_to_send = min(self.batch_size, len(messages))
+                        # Move batch from messages to batch 
+                        batch = messages[0:count_to_send]
+                        del messages[0:count_to_send]
+                        # Send batch
+                        self.send_socket(b''.join(batch))
```

Comment:
This is definitely better, but I think it's still an unusual way to produce the batches: using `del` in Python is rare, and it's simpler and more efficient to just slice the batch you need (as it avoids the second copy-down for the `del`). So I would much prefer the `for` loop like I mentioned above.

Additionally, with the for loop we can get rid of the `if messages:` block and just have the `for` loop block -- if there are no messages it'll work the same way.